File: Algorithms/160. Intersection of Two Linked Lists/golang/solution.hash-table.go
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */
func getIntersectionNode(headA, headB *ListNode) *ListNode {
	var nodesInB = make(map[*ListNode]bool)

	for headB != nil {
		nodesInB[headB] = true
		headB = headB.Next
	}

	for headA != nil {
		if nodesInB[headA] {
			return headA
		}
		headA = headA.Next
	}
	return nil
}

File: Algorithms/160. Intersection of Two Linked Lists/javascript/solution.hash-table.js
/**
 * Definition for singly-linked list.
 * function ListNode(val) {
 *     this.val = val;
 *     this.next = null;
 * }
 */

/**
 * @param {ListNode} headA
 * @param {ListNode} headB
 * @return {ListNode}
 */
var getIntersectionNode = function (headA, headB) {
    const nodesInB = new Set();

    while (headB) {
        nodesInB.add(headB);
        headB = headB.next;
    }
    while (headA) {
        if (nodesInB.has(headA)) {
            return headA;
        }
        headA = headA.next;
    }
    return null;
};
